### DISCUSSION TITLE
[RFC] Improve query UX

### DISCUSSION THREAD

**Discussion by Janpot** [mui/toolpad#1752] [Early feedback]
### What's the problem? 🤔

The following options

![image](https://user-images.githubusercontent.com/2109932/225294999-dbac7334-c5f9-40d7-9cf8-5de9b99faafa.png)

1. Query fetch mode: The default is at "execute automatically" which makes most sense for displaying data, but this option is very passive in a secondary location, one needs to conscientiously change it when configuring non-idempotent queries. This is often forgotten and upon clicking **Save** the page automatically starts fetching the data.
2. Enabled. This is only relevant for automatic queries. The prime use-case was to disable the query under certain conditions. Usually when query input isn't available yet.
3. Refetch interval: There is no clear use-case for this
4. Parameters: hard to test and preview queries with parameter values. Especially when they are bound.
5. Configurable cache time: https://github.com/mui/mui-toolpad/issues/1093

### What are our options? 💡

1. Changing this will be about making sure that users are fully aware the query starts fetching automatically upon clicking "save". There are multiple directions we can go with this
    - Add a modal selection between the two modes. i.e. before the user can start configuring the query they must choose a mode.
    - Split the two modes again. Reserve "query" for bringing data on the page (the automatic mode) and come up witha  different way to define the non-idempotent requests
    - Try to detect from the query configuration whether it's idempotent or not
2. The prime intention was to make sure parameters are initialized before making requests. Perhaps we could add a "required" option to parameters along with a "default value". All required parameters must then have a value before the query starts fetching automatically.
3. We can remove this until clear use-cases become apparant. e.g. one use case could be to auto-reload dashboards at intervals. But perhaps a page-wide setting makes more sense in that case?
4. Design a parameters UI which allows for the following features for parameters:
    - Set a default value
    - Bind to state on the page
    - Set secondary options, e.g. "required"
    - Set temporary values for testing/preview

**Comment by apedroferreira** [comment]
1. If the issue could be just about splitting queries and mutations, I feel like we could maybe make them different concepts. And unless mistakenly "automatic" queries are causing many issues to users, I feel like this should be the default for queries without requiring an extra step? But I might be missing some things.
2. Covering this with default values seems like a good way to do this at first glance. Some kind of `skip` expression that returns a boolean could also cover this issue, but while more powerful, it would also be more complex and difficult to implement well.
3. Fine with removing it
